internal/services/instances.go

```diff
@@ -178,6 +210,195 @@ func (s *ServiceClient) ListInNamespace(ctx context.Context, namespace string) (
+			"%s,%s",
+			ServiceNameLabelKey,
+			CatalogServiceLabelKey,
+		)
+	}
+
+	services, err := s.kubeClient.Kubectl.CoreV1().Secrets(namespace).List(ctx, listOpts)
+
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			return serviceList, nil
+		}
+		return nil, errors.Wrap(err, "listing the service instances")
+	}
+
+	catalogServices, err := s.ListCatalogServices(ctx)
+	if err != nil {
+		return nil, errors.Wrap(err, "error getting catalog services")
+	}
+
+	// catalogServiceNameMap is a lookup map to check the available Catalog Services
+	catalogServiceNameMap := map[string]struct{}{}
+	for _, catalogService := range catalogServices {
+		catalogServiceNameMap[catalogService.Meta.Name] = struct{}{}
+	}
+
+	for _, srv := range services.Items {
+		catalogServiceName := srv.GetLabels()[CatalogServiceLabelKey]
+		if _, exists := catalogServiceNameMap[catalogServiceName]; !exists {
+			catalogServiceName = "[Missing] " + catalogServiceName
+		}
+
+		serviceName := srv.GetLabels()[ServiceNameLabelKey]
+
+		service := models.Service{
+			Meta: models.Meta{
+				Name:      serviceName,
+				Namespace: srv.ObjectMeta.Namespace,
+				CreatedAt: srv.GetCreationTimestamp(),
+			},
+			CatalogService:        catalogServiceName,
+			CatalogServiceVersion: srv.GetLabels()[CatalogServiceVersionLabelKey],
+		}
+
+		logger := tracelog.NewLogger().WithName("ServiceStatus")
+		serviceStatus, err := helm.Status(ctx, logger, s.kubeClient,
+			srv.ObjectMeta.Namespace, names.ServiceReleaseName(serviceName))
+		if err != nil {
+			if errors.Is(err, helmdriver.ErrReleaseNotFound) {
+				serviceStatus = "Not Ready" // The installation job is still running?
+			} else {
+				return nil, errors.Wrap(err, "finding helm release status")
+			}
+		}
+
+		service.Status = models.NewServiceStatusFromHelmRelease(serviceStatus)
+
+		serviceList = append(serviceList, service)
+	}
+
+	// COMPATIBILITY SUPPORT - List (helm controller)-based services too.
+	serviceListHC, err := s.listHC(ctx, namespace)
+	if err != nil {
+		return nil, errors.Wrap(err, "listing the (helm controller)-based service instances")
+	}
+
+	return append(serviceList, serviceListHC...), nil
+}
+
+func serviceResourceName(name string) string {
+	return names.GenerateResourceName("s", name)
+}
+
+// -----------------------------------------------------------------------------------------------
+// COMPATIBILITY SUPPORT for services from before https://github.com/epinio/epinio/issues/1704 fix
+//
+// This is essentially all of the old Get/Delete(All)/List* functions, renamed with an added `HC`
+// suffix (for Helm Controller). The new functions run them in appropriate places.
+//
+// NOTE that `Create` is NOT in this list. We do not create (helm controller)-based services anymore.
+//
+
+// GetHC returns a Service "instance" object if one is exist, or nil otherwise.  Also returns an
+// error if one occurs.
+func (s *ServiceClient) GetHC(ctx context.Context, namespace, name string) (*models.Service, error) {
```

Comment:
I would avoid shortening too much func names using acronyms. Something like `GetHelmChartService` is clearer.

Reply:
Fixed. Commit 635b0a7778fc6a5afea6d9dd8f0160ef07f86092